cmd/query/app/querysvc/query_service_v2_test.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package querysvc
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/storage_v2/depstore"
+	depstoremocks "github.com/jaegertracing/jaeger/storage_v2/depstore/mocks"
+	"github.com/jaegertracing/jaeger/storage_v2/tracestore"
+	tracestoremocks "github.com/jaegertracing/jaeger/storage_v2/tracestore/mocks"
+)
+
+const millisToNanosMultiplier = int64(time.Millisecond / time.Nanosecond)
+
+var (
+	defaultDependencyLookbackDuration = time.Hour * 24
+
+	testTraceID = pcommon.TraceID([16]byte{1})
+)
+
+type testQueryServiceV2 struct {
+	queryService *QueryServiceV2
+	traceReader  *tracestoremocks.Reader
+	depsReader   *depstoremocks.Reader
+
+	archiveTraceReader *tracestoremocks.Reader
+	archiveTraceWriter *tracestoremocks.Writer
+}
+
+type testOptionV2 func(*testQueryServiceV2, *QueryServiceOptionsV2)
+
+func withArchiveTraceReader() testOptionV2 {
+	return func(tqs *testQueryServiceV2, options *QueryServiceOptionsV2) {
+		r := &tracestoremocks.Reader{}
+		tqs.archiveTraceReader = r
+		options.ArchiveTraceReader = r
+	}
+}
+
+func withArchiveTraceWriter() testOptionV2 {
+	return func(tqs *testQueryServiceV2, options *QueryServiceOptionsV2) {
+		r := &tracestoremocks.Writer{}
+		tqs.archiveTraceWriter = r
+		options.ArchiveTraceWriter = r
+	}
+}
+
+func initializeTestServiceV2(optionAppliers ...testOptionV2) *testQueryServiceV2 {
+	traceReader := &tracestoremocks.Reader{}
+	dependencyStorage := &depstoremocks.Reader{}
+
+	options := QueryServiceOptionsV2{}
+
+	tqs := testQueryServiceV2{
+		traceReader: traceReader,
+		depsReader:  dependencyStorage,
+	}
+
+	for _, optApplier := range optionAppliers {
+		optApplier(&tqs, &options)
+	}
+
+	tqs.queryService = NewQueryServiceV2(traceReader, dependencyStorage, options)
+	return &tqs
+}
+
+func TestGetServicesV2(t *testing.T) {
+	tqs := initializeTestServiceV2()
+	expectedServices := []string{"trifle", "bling"}
+	tqs.traceReader.On("GetServices", mock.AnythingOfType("*context.valueCtx")).Return(expectedServices, nil).Once()
+
+	type contextKey string
+	ctx := context.Background()
+	actualServices, err := tqs.queryService.GetServices(context.WithValue(ctx, contextKey("foo"), "bar"))
+	require.NoError(t, err)
+	assert.Equal(t, expectedServices, actualServices)
+}
+
+func TestGetOperationsV2(t *testing.T) {
+	tqs := initializeTestServiceV2()
+	expectedOperations := []tracestore.Operation{{Name: "", SpanKind: ""}, {Name: "get", SpanKind: ""}}
+	operationQuery := tracestore.OperationQueryParameters{ServiceName: "abc/trifle"}
```

Comment:
we need to rename this struct, to match `GetTraceParams`, `TraceQueryParams`:
```suggestion
	operationQuery := tracestore.OperationQueryParams{ServiceName: "abc/trifle"}
```